packages/collector/src/announceCycle/agentHostLookup.js

```diff
@@ -166,6 +164,51 @@ function checkHost(host, cb) {
   req.end();
 }
 
+/**
+ * Checks the response payload to determine whether we have actually connected to an Instana host agent.
+ *
+ * @param {string} host the target IP (will only be used for logging in this method)
+ * @param {import('http').IncomingMessage} res the HTTP response
+ * @param {(...args: *) => *} cb
+ */
+function handleResponse(host, res, cb) {
+  // We inspect the payload, checking whether it is JSON and has the "version" property, to verify that we have
+  // actually connected to an Instana host agent.
+
+  /** @type {Array.<Buffer>} */
+  const chunks = [];
+  res.on('data', chunk => {
+    chunks.push(chunk);
+  });
+  res.on('end', () => {
+    const responsePayload = String(Buffer.concat(chunks));
+    let responseJson;
+    try {
+      responseJson = JSON.parse(responsePayload);
+    } catch (e) {
+      cb(
+        new Error(
+          `The attempt to connect to the Instana host agent on ${host}:${agentOpts.port} has failed, the response ` +
+            `cannot be parsed as JSON. The party listening on ${host}:${agentOpts.port} does not seem to be an ` +
+            `Instana host agent. Full response: ${responsePayload}`
+        )
+      );
+      return;
+    }
+    if (responseJson.version !== undefined) {
```

Comment:
Suggestion: Add a comment saying that this check is a "non official/professional/temporarly way" to figure out if the host might be the agent.

Reply:
It is quite "official" though: https://github.ibm.com/instana/technical-documentation/blob/master/tracing/specification/README.md#phase-1--host-lookup. I'll add a link to that part of the spec here instead.

Reply:
... that is, I'll add it to the existing comment at the start of the method: https://github.com/instana/nodejs/pull/794/files#diff-44f44550231a95107349e86cfa335b336d7205479f2422abb87cc6386d4b882bR175-R176